couchcopy

```diff
@@ -37,6 +37,20 @@ async def subprocess(*args, shell=False):
             stderr=asyncio.subprocess.PIPE)
     stdout, stderr = await p.communicate()
     stdout, stderr = stdout.decode(), stderr.decode()
+    if args[0] == 'rsync' and p.returncode == 24:
+        # When a file is deleted between the start of rsync and the moment
+        # the file is copied, rsync output a "warning: some files vanished" log
+        # and it exits with return code 24.
+        # See upstream explanation:
+        # https://bugzilla.samba.org/show_bug.cgi?id=10356
+        # Despite the non-zero return code, rsync does its work as if
+        # nothing happens.
+        # We think it's OK that CouchDB delete files thus we print the rsync
+        # warning but we ignore its return code.
+        # If more critical exit codes are returned, they are prioritized by
+        # rsync, so it's OK to ignore exit code 24.
+        print(stderr)
```

Comment:
Is this even needed to output something alarming on the console, while we know that it's OK?

Reply:
The message outputed is:
```
file has vanished: "/tmp/couchcopy-8b5iag9h/data/shards/00000000-7fffffff/delete_me"
rsync warning: some files vanished before they could be transferred (code 24) at main.c(1330) [sender=3.2.3]
```
I think it's not that scary, and maybe it could help people debug stuff?

Reply:
OK! In this case I agree with you, it's not as scary as I imagined (and by the way it was shown in PR notes, which I didn't read).